include/libp2p/protocol/gossip/impl/gossip_wire_protocol.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_GOSSIP_WIRE_PROTOCOL_HPP
+#define LIBP2P_GOSSIP_WIRE_PROTOCOL_HPP
+
+#include <libp2p/protocol/gossip/common.hpp>
+
+#include <map>
+
+#include <gsl/span>
+
+namespace pubsub::pb {
+  class RPC;
+  class ControlMessage;
+}  // namespace pubsub::pb
+
+namespace libp2p::protocol::gossip {
+
+  /// Subscribe-unsubscribe request
+  struct SubOpts {
+    /// true means subscribe, false means unsubscribe
+    bool subscribe = false;
+
+    /// Pub-sub topic
+    TopicId topic_id;
+  };
+
+  /// Announces about topic messages available on a host
+  using IHaveTable = std::map<TopicId, Repeated<MessageId>>;
+
+  /// General wire protocol message
+  struct RPCMessage {
+    /// Subscribe-unsubscribe requests
+    Repeated<SubOpts> subscriptions;
+
+    /// "I have" announces by topics
+    IHaveTable i_have;
+
+    /// Requests for messages
+    Repeated<MessageId> i_want;
+
+    /// Requests to join per-topic mesh
+    Repeated<TopicId> graft;
+
+    /// Requests to leave per-topic mesh
+    Repeated<TopicId> prune;
+
+    /// Messages to publish
+    Repeated<TopicMessage::Ptr> publish;
+
+    bool empty() const;
+
+    void clear();
+
+    bool deserialize(gsl::span<const uint8_t> bytes);
+
+    bool serialize(std::vector<uint8_t> &buffer) const;
+  };
+
+  /// Interface for accepting sub messages being read from wire
+  class MessageReceiver {
```

Comment:
Please use the same name for header file and class that it contains. 
Why this file is named gossip_wire_protocol?